src/mesh/exodusII_io.C

```diff
@@ -286,7 +286,30 @@ void ExodusII_IO::read (const std::string & fname)
 
           // Assign extra integer IDs
           for (auto & id : extra_ids)
-            elem->set_extra_integer(id, std::round(elem_ids[id][elem->id()]));
+          {
+            Real v = elem_ids[id][elem->id()];
+            if (std::numeric_limits<Real>::digits > std::numeric_limits<dof_id_type>::digits)
```

Comment:
No, no - we don't want to compare float/double/quad to int/long here, we want to see if v, specifically, is in the range of numbers that can be exactly represented as a double.

Reply:
Oh, wait, I get it now - you're trying to short circuit entirely in the case where we have a large Real but a small dof_id_type.  Disregard the above - that's clever!

Reply:
;-) yes, otherwise the bit shift operation could create some strange behavior.